### PR TITLE
Made UI responsive

### LINKED ISSUE #104: Profile screen crashes on first login
To reproduce the issue:
1. Logout the current user and close the app
2. Reopen the app and login
3. Now go to profile screen

image

### PR DESCRIPTION
**Pull request by Aarush-Acharya**
Took care of all the widgets using ratios via Get.height and Get.width. Made the UI generic and responsive to adapt to almost all the screen sizes

closes #104 

Screenshot 2023-08-03 at 2 00 43 AM

### CODE REVIEW

**Comment by Aarush-Acharya** [comment]
Hey @chandansgowda I am done with the changes that were to be made by me

**Review by chandansgowda — APPROVED** [review]
Great job @Aarush-Acharya

File: lib/views/screens/home_screen.dart [hunk 1]
```diff
@@ -20,10 +20,10 @@ class HomeScreen extends StatelessWidget {
               child: CustomRefreshIndicator(
                   builder: MaterialIndicatorDelegate(
                     builder: (context, controller) {
-                      return const Icon(
+                      return  Icon(
                         Icons.ac_unit,
                         color: Colors.amber,
-                        size: 30,
+                        size: 0.01825*Get.height+0.0364*Get.width,
```
Comment (jddeep) [inline lib/views/screens/home_screen.dart:26]: how are we coming with these random ratios ?

File: lib/views/screens/login_screen.dart [hunk 2]
```diff
@@ -70,6 +74,9 @@ class LoginScreen extends StatelessWidget {
                               Icons.lock_outline_rounded,
                             ),
                             labelText: "Password",
+                            errorStyle: TextStyle(
+                                fontSize:
+                                    0.0085 * Get.height + 0.017 * Get.width),
                             suffixIcon: IconButton(
```
Comment (jddeep) [inline lib/views/screens/login_screen.dart:80]: same with these ratios ?

File: lib/views/screens/room_screen.dart [hunk 3]
```diff
@@ -46,42 +46,43 @@ class _RoomScreenState extends State<RoomScreen> {
             children: [
               Text(
                 widget.room.name,
-                style: TextStyle(fontSize: 20, color: Colors.amber),
+                style: TextStyle(fontSize: 0.012*Get.height+0.0243*Get.width, color: Colors.amber),
```
Comment (jddeep) [inline lib/views/screens/room_screen.dart:49]: these as well. Please try not to use such constant values.

File: lib/views/screens/home_screen.dart (hunk 1 above)
Comment (Aarush-Acharya) [reply lib/views/screens/home_screen.dart:26]: Well, the size of the icon needed to be changed not just by the width or the height has to take into account both of them so in order to make the UI accurately responsive I divided sizes for Icons and Texts into two equal parts and converted one part into the ratio to the height (the values are ratio to the height of the screen at which everything looked perfect)  while other into ratio to the width of the ideal/original screen. 

Example:  
30/2 = 15,
15/821.7 (height of the original screen) = 0.01825 
15/411.4 (width of the original screen) = 0.0364

Hence the new representation of 30:  0.01825*Get.height + 0.0364*Get.width

This way both height and width can equally contribute to these sizes did this for all the Text and Icon sizes throughout the UI

File: lib/views/screens/login_screen.dart (hunk 2 above)
Comment (Aarush-Acharya) [reply lib/views/screens/login_screen.dart:80]: Here I converted the default text size to dynamic to attain a precision of responsiveness default text size in Dart is 14 

so 7 goes for the height and 7 for the width 
7/411.4 (width of the original screen) = 0.017...
7/821.7 (height of the original screen) = 0.0085....

File: lib/views/screens/room_screen.dart (hunk 3 above)
Comment (Aarush-Acharya) [reply lib/views/screens/room_screen.dart:49]: Did this to make the static size 20 into dynamic with respect to height and width again divided 20 into 10 for height  and 10 for width and applied the stated methodology

**Comment by jddeep** [comment]
how can you say height of every screen the app is opened is 821 and width is 411 ? @Aarush-Acharya

**Comment by Aarush-Acharya** [comment]
Well the ratios are extracted from the height and width of the original screen and then 
`Get.height` returns the height of the current screen 
`Get.width` returns the width of the current screen

**Comment by jddeep** [comment]
@Aarush-Acharya then put those two values in the constants file and use from there everywhere. It will be difficult to understand otherwise.
@chandansgowda have you tested this across different devices (all possible) ?

**Comment by chandansgowda** [comment]
@jddeep I have tested it on three of my virtual devices. Can you please suggest ways to test it on all devices ?

**Comment by Aarush-Acharya** [comment]
@jddeep would do so also rather than just the constant values shall I put the whole expression `0.012*Get.height+0.0243*Get.width` in constant files 
would suggest by the nomenclature and comment what size that expression represents and would just use different sizes from here across the application

Also, I have a prior commitment till the 7th morning would be able to work later that morning

**Comment by jddeep** [comment]
@Aarush-Acharya second way is fine. Just put the values `821` and `411` in a constant file and use from there everywhere.

**Comment by chandansgowda** [comment]
@Aarush-Acharya Better if you can create a new file called ui_constants.dart and put everything there.

**Comment by jddeep** [comment]
@Aarush-Acharya @chandansgowda Please resolve conflicts

**Comment by chandansgowda** [comment]
> @Aarush-Acharya @chandansgowda Please resolve conflicts

@jddeep He's still working on the PR.

**Comment by Aarush-Acharya** [comment]
@jddeep @chandansgowda apologies for the delay will commit it by EOD tomorrow

**Comment by Aarush-Acharya** [comment]
@jddeep @chandansgowda apologies for the delay will commit it by EOD

**Comment by jddeep** [comment]
@Aarush-Acharya Please resolve conflicts

**Comment by Aarush-Acharya** [comment]
Alright on it

**Comment by Aarush-Acharya** [comment]
To be cleaned

- [ ] create_room_screen.dart
- [ ] discussions_screen.dart
- [ ] email_verification_screen.dart
- [X] home_screen.dart
- [X] login_screen.dart
- [ ] onboarding_screen.dart
- [ ] profile_screen.dart
- [ ] room_screen.dart
- [ ] signup_screen.dart
- [ ] splash_screen.dart
- [ ] tabview_screen.dart

**Review by jddeep — APPROVED** [review]

**Comment by Aarush-Acharya** [comment]
Hey @jddeep this PR was still underwork 2 screen have been built and improvements were on the way shall I create a new PR for the rest of the updates

**Pull request merged**